dataRetriever/requestHandlers/requestHandler.go

```diff
@@ -196,7 +197,8 @@ func (rrh *resolverRequestHandler) RequestRewardTransactions(destShardID uint32,
 
 // RequestMiniBlock method asks for miniblock from the connected peers
 func (rrh *resolverRequestHandler) RequestMiniBlock(destShardID uint32, miniblockHash []byte) {
-	if !rrh.testIfRequestIsNeeded(miniblockHash) {
+	suffix := fmt.Sprintf("miniblocks_%d", destShardID)
```

Comment:
I think smaller suffix would help.

Reply:
Like 2 letters - and maybe define them as costs

Reply:
on it

Reply:
done